### PR TITLE
Enable user defined and JSBSim Magnetometer XML Input Mapping

### PR DESCRIPTION
**Pull request by mvacanti**
**Proposal Description**

- Enable the user to map JSBSim variables to the bridge input properties via airframe configuration files as per previous.
- Provide an example model of varying magnetometer bias based on throttle input in px4_default_mag_sensor.xml.
- Does not modify the default behavior of any airframe
- **NOTE:** Stable JSBSim magnetometer startup behavior dependent on https://github.com/JSBSim-Team/jsbsim/issues/332

**To Test**
- Build separate branch of JSBSim: https://github.com/mvacanti/jsbsim/tree/pr-update-mag-startup 
- Modify configs/quadrotor.xml to map desired variables
- Modify models/quadrotor.xml to include default_px4_mag_sensor.xml system file

### CODE REVIEW

File: src/sensor_mag_plugin.cpp [hunk 1]
```diff
@@ -122,4 +144,4 @@ void SensorMagPlugin::addNoise(Eigen::Vector3d* magnetic_field, const double dt)
     _bias[i] = phi_d * _bias[i] + sigma_b_d * _standard_normal_distribution(_random_generator);
     (*magnetic_field)[i] = (*magnetic_field)[i] + _bias[i] + sigma_d * _standard_normal_distribution(_random_generator);
   }
-}
+}
```
Comment (Jaeyoung-Lim) [inline src/sensor_mag_plugin.cpp:147]: Could you add a EOF just to make it more clean?

**Review by Jaeyoung-Lim — APPROVED** [review]
Thanks, looks good to me!

Is this still WIP? or just marked as a draft?

File: src/sensor_mag_plugin.cpp (hunk 1 above)
Comment (mvacanti) [reply src/sensor_mag_plugin.cpp]: Done.

**Comment by mvacanti** [comment]
I am done with my work, are you OK with the way I handled the open issue with JSBSim?

**Comment by Jaeyoung-Lim** [comment]
@mvacanti Yes, but does this mean that it would fail for other jsbsim distributions?

**Comment by mvacanti** [comment]
@Jaeyoung-Lim on the current JSBSim release it does not fail outright when using the magnetometer system file, for the first several seconds it creates an EKF magnetometer error that prevents arming due to erroneous startup data followed by a step-change. Once that clears it works fine. I have a PR for JSBSim that will be rolled into the next [bug-fix release](https://github.com/JSBSim-Team/jsbsim/pull/334). The default behavior (no magnetometer JSB system) is unaffected by either.

**Comment by Jaeyoung-Lim** [comment]
@mvacanti Thanks, looks like your fix went into jsbsim! merging!

**Pull request merged**